### ISSUE TITLE
New Feature: support Mode Hanzi with GB2312 subset

### ISSUE THREAD

**Issue by neycyanshi** [heuer/segno#74]
Current zxing can [decode Hanzi segment](https://github.com/zxing/zxing/blob/0cf3b9be71680f50c90a71ca26ce0d33664b0dd6/core/src/main/java/com/google/zxing/qrcode/decoder/DecodedBitStreamParser.java#L99-L107). See specification GBT 18284-2000. 
But I cannot find any QR code generator that support this CodecMode. It would be great if segno can encode and generate it.

**Comment by heuer** [comment]
Thanks for your suggestion.

Is the specification available somewhere?

I'd be happy to help you but I have no clue where to start. I searched for "python hanzi" to get some examples but I wasn't able to detect any useful hints.

**Comment by neycyanshi** [comment]
I have added the Mode Hanzi with GB2312 sub-character set here #75 .
Also, there is a [bug in Mode Kanji on the return length](https://github.com/heuer/segno/blob/4ea7de580cb5b5552fb0501d26f15c0cace668be/segno/encoder.py#L1010). Each Kanji/Hanzi character is represented by 2 bytes code in GB2312 standard. So the length of character should be the length of string, or the length of data divided by 2.

**Comment by heuer** [comment]
Thanks! I'll take look soon! Kanji mode: Good catch!

**Comment by heuer** [comment]
Your contribution #75 suggests a problem with Kanji mode, but on closer inspection I am nearly convinced that Segno is currently implementing the Kanji mode correctly. I don't see a difference between the current code and your code. Your code tries to determine the length beforehand, but in my opinion there are no changes to my code, which determines the length at the end.

A test case that shows a problem would be helpful.

The Kanji problem may also need to be separated from your suggestion for the Hanzi mode.

**Comment by neycyanshi** [comment]
I move the len() before encoding, because the encoding will change the length of `data`.
For example:
```
data='漢字'
len(data)  # = 2  
data = data.encode('shift_jis')  
len(data)  # = 4
```
From specification, the return value should be the character count, which is 2 as above.
As you can see, the 1st QR below is the QR generated before this fix, and the 2nd after fix. You could scan with any QR scanner, only the 2nd QR can be scanned.
![Kanji1](https://user-images.githubusercontent.com/16794154/88446692-a6dd7380-ce5e-11ea-9ef4-deb1a7e784ec.png)
![Kanji2](https://user-images.githubusercontent.com/16794154/88446694-a9d86400-ce5e-11ea-897b-0fcfdc673265.png)


I want to separate this `fix` with `Hanzi mode`, but they have to be together to generate a correct QR. So I keep them together.

**Comment by heuer** [comment]
> I move the len() before encoding, because the encoding will change the length of `data`.

Thanks and thanks for the example as well! Yes, you're right.

Fixed within release 1.0.1

I separated that from the Hanzi feature, I need to write some test cases for it.

**Comment by neycyanshi** [comment]
Wait a minute, you only have to change char count in `data_to_bytes()`.
Don't change the iteration upperbound here in `make_segment()`! You are only using half data now.
https://github.com/heuer/segno/blob/45d91f79649f49440b72449d2a773c8a85a8e57e/segno/encoder.py#L1079-L1080

**Comment by heuer** [comment]
> Don't change the iteration upperbound here in `make_segment()`! You are only using half data now.

Fixed with cc169235127a6f83e610f26b9f330ce510244e4b 

See release [1.0.2](https://pypi.org/project/segno/1.0.2/) and [tag 1.0.2](https://github.com/heuer/segno/releases/tag/1.0.2)

**Comment by heuer** [comment]
Thanks for your patience and work. Segno 1.1.0 (just published) supports Hanzi
